wordpress-mobile/WordPress-iOS
# Workaround to support ExPlat in Jetpack app

Ref p1682352977248619-slack-C04M1NX8YCR

## Description

This PR does the following:
- Updates `Automattic-Tracks-iOS` library from `0.13` to `2.0.0+`.
- Tweak ExPlat configuration to make it work for Jetpack. But there are no experiments to test in this PR yet.

## Related PRs

**This PR is blocked By:**
- https://github.com/Automattic/Automattic-Tracks-iOS/pull/253

**And blocking:**
- https://github.com/wordpress-mobile/WordPress-iOS/pull/20590

## Test Instructions

>Updates `Automattic-Tracks-iOS` library from `0.13` to `2.0.0+`.

Because of the change, we should double check Track events are working as expected.

1. Set a breakpoint in `TracksService.m` line 141.
```objective-c
- (void)trackEventName:(NSString *)eventName withCustomProperties:(NSDictionary *)customProperties
{
    eventName = [NSString stringWithFormat:@"%@_%@", self.eventNamePrefix, eventName];
    
🔵  [self.tracksEventService createTracksEventWithName:eventName
                                              username:self.username
                                                userID:self.userID
```
2. Build and run the Jetpack / WordPress app.
3. Navigate to any screen until the app is paused.
4. Head to the debugger and type `po eventName`.
5. **Expect** the event name to be logged and starts with `jpios` or `wpios` depending on whether the running app is Jetpack or WordPress.

## Regression Notes
1. Potential unintended areas of impact
`Automattic-Tracks-iOS` pod was updated from `0.13` to `2.0.0`, which is a significant update. We should smoke test whether track events are working as expected. See Test Instructions.

6. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual Testing.

7. What automated tests I added (or what prevented me from doing so)
None.

## PR submission checklist:

UI Changes testing checklist:
